Decorative Flags (also handheld flags)

SaveliyM360 opened the pull request:
<!-- Please read these guidelines before opening your PR: https://docs.spacestation14.io/en/getting-started/pr-guideline -->
<!-- The text between the arrows are comments - they will not be visible on your PR. -->

## About the PR
I added flags to the game. It's a nice addition to the decor, and also allows servers to add their own faction and country lore flags, thanks to the existing blank flag base (hello from Baystation), such as Transsolar Federation, Space Wizard Federation, Moghes State, Arachnid Empire... I think you get the idea.

Of the variations of flags, there is the flag of NT, Syndicate, Pirates, LGBT, various color variations, as well as handheld (handy) versions for each of the flags.

## Why / Balance
I wrote above, and it's also not cool that there are banners in the game, but there are no flags.

## Technical details
They can be breaked just like posters. But unfortunately, they are available so far only in sandbox mode for mapping. Maybe in the future I will add them to the station maps (I hope it won't break anything).
Also, handy flags can be taken from the sandbox, and so far cannot be obtained in the game, as the big versions.

**_How to create your flag based on a blank flag sprite:_** Take the blank flag sprite, create a new layer under it, and set the blending mode to "multiple" on the blank flag layer. Before that, I recommend to select and cut on a new layer the sprites of "nails", which are located on the upper edges of the flag, and look like a "plus". 
(It work on paint.net, but you can check it yourself in Photoshop)

## Media
<!-- 
PRs which make ingame changes (adding clothing, items, new features, etc) are required to have media attached that showcase the changes.
Small fixes/refactors are exempt.
Any media may be used in SS14 progress reports, with clear credit given.

If you're unsure whether your PR will require media, ask a maintainer.

Check the box below to confirm that you have in fact seen this (put an X in the brackets, like [X]):
-->
![Flags_new2](https://github.com/space-wizards/space-station-14/assets/142160914/71b55c5a-8ac3-4ef2-b021-56fa085ce1e1)



- [X] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

**Changelog**
<!--
Make players aware of new features and changes that could affect how they play the game by adding a Changelog entry. Please read the Changelog guidelines located at: https://docs.spacestation14.io/en/getting-started/pr-guideline#changelog
-->
- add: Added a large number of wall flags.  Now you can feel more patriotic by hanging your faction's flag on the wall.
- add: Added handheld flags in addition to large flags. Wave them around so everyone can see what a patriot you are.
<!--
Make sure to take this Changelog template out of the comment block in order for it to show up.
:cl:
- add: Added fun!
- remove: Removed fun!
- tweak: Changed fun!
- fix: Fixed fun!
-->

lzk228 commented on Resources/Prototypes/Entities/Structures/Wallmounts/Signs/flags.yml:
```diff
@@ -0,0 +1,37 @@
+  parent: BaseFlag
+  id: SyndieFlag
+  name: Syndicate flag
+  description: "Smells bloody. Fuck NT!"
```
that's a bad word

CatMagic1 replied:
merge blocked

deltanedas commented:
real

Emisse commented on Resources/Textures/Structures/Wallmounts/flags.rsi/meta.json:
```diff
@@ -0,0 +1,20 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Taken from tgstation at https://github.com/tgstation/tgstation/commit/d0d81185f09ca30d3b0856d476544240dba0de53, drawed by Saveliy360",
```
```suggestion
  "copyright": "Taken from tgstation at https://github.com/tgstation/tgstation/commit/d0d81185f09ca30d3b0856d476544240dba0de53, drawn by Saveliy360",
```

ficcialfaint commented on Resources/Textures/Structures/Wallmounts/flags.rsi/meta.json:
```diff
@@ -0,0 +1,20 @@
+{
+  "version": 1,
+  "license": "CC-BY-SA-3.0",
+  "copyright": "Taken from tgstation at https://github.com/tgstation/tgstation/commit/d0d81185f09ca30d3b0856d476544240dba0de53, drawn by Saveliy360",
```
doesn't seem too clear: what exactly was drawn by you and what was taken from /tg/

ficcialfaint commented on Resources/Prototypes/Entities/Structures/Wallmounts/Signs/flags.yml:
Can it spawn a destructed version of the flag as posters do?

JoeHammad1844 commented:
you forgot to mention you also added the french flag

mirrorcult commented on Resources/Prototypes/Entities/Objects/Misc/handy_flags.yml:
```diff
@@ -0,0 +1,167 @@
+    
+- type: entity
+  parent: BaseItem
+  id: BlueHandyFlag
+  name: blue handy flag
+  description: "Just black handy flag"
```
description is incorrect

mirrorcult commented on Resources/Prototypes/Entities/Objects/Misc/handy_flags.yml:
```diff
@@ -0,0 +1,167 @@
+    
+- type: entity
+  parent: BaseItem
+  id: PirateHandyFlag
+  name: Pirate handy flag
+  description: "Holding it in your hands, show these carp that you're not kidding."
```
```suggestion
  id: PirateHandyFlag
  name: pirate handy flag
  description: Hold it in your hands to show these carp that you're not kidding.
```
descs dont need quotation marks around them + dont capitalize pirate imo
descriptions for most of these read pretty weirdly imo but ill just point them out as i go

mirrorcult commented on Resources/Textures/Structures/Wallmounts/flags.rsi/meta.json:
```diff
@@ -0,0 +1,53 @@
+    {
+      "name": "syndie_flag"
+    },
+	{
+      "name": "lgbtq_flag"
+    },
+	{
+      "name": "pirate_flag"
+    },
+	{
+      "name": "black_flag"
+    },
+	{
```
lots of mixed tabs & spaces in this file, please only use spaces

mirrorcult commented on Resources/Prototypes/Entities/Objects/Misc/handy_flags.yml:
```diff
@@ -0,0 +1,167 @@
+- type: entity
+  parent: BaseItem
+  id: LGBTQHandyFlag
+  name: LGBTQ handy flag
+  description: "Be gay do crime handy flag"
```
```suggestion
  name: LGBTQ handy flag
  description: "The be gay do crime handy flag."
```

all flag descs here need periods at the end of them, the solid color flags mostly

also think this desc reads better like this, same for the wallmount flag

mirrorcult commented on Resources/Prototypes/Entities/Objects/Misc/handy_flags.yml:
```diff
@@ -0,0 +1,167 @@
+    
+- type: entity
+  parent: BaseItem
+  id: BlackHandyFlag
+  name: black handy flag
+  description: "Just black handy flag"
```
```suggestion
  id: BlackHandyFlag
  name: black handy flag
  description: Just a black handy flag.
```

same for all the other color flags, not going to review them separately

mirrorcult reviewed (changes requested):
the solid color flags can very easily use a grayscale layer that has its color specified in the sprite component instead, rather than duplicating so many images like this. the other ones are fine to be unique. (maybe even just try setting color on the gray solid color flag and see how it looks?)

"handy flag" reads kind of weird to me? ive never heard them called that before in english, i would call them "handheld flags"

SaveliyM360 commented:
> Ok, I think I fixed all the typos in the files, and renamed all the handy things to handheld. About grayscale mechanics for the flags, I don't really understand it. When I creating colored flags, I focused on the color variety of bedspreads, because I thought that five flags would be too few.
If you mean to remove the colored flags, then I will.

mirrorcult commented:
you can just have one grayscaled sprite for the flags, and then color them dynamically in YAML by changing the `color:` field in spritecomponent

SaveliyM360 commented:
> you can just have one grayscaled sprite for the flags, and then color them dynamically in YAML by changing the `color:` field in spritecomponent

Oh, i got it. Well, in fact, this role is performed by the blank flag.

SaveliyM360 commented:
Okay, now all coloured flags are removed.

SaveliyM360 commented:
I replaced the picture, due to the removal of colored flags.

metalgearsloth reviewed (changes requested):
"Drawn by" not "drawed by" please, also faint's review.
